spec/Candid.md

```diff
@@ -711,7 +905,33 @@ not (null <: <datatype>)
 <datatype> <: <datatype'> ~> f
 ------------------------------------------
 <datatype> <: opt <datatype'> ~> \x.?(f x)
+
+not (<datatype> <: <datatype'>)
+---------------------------------
+opt <datatype> <: opt <datatype'>
+  ~> \x.join_opt (\y.
+       if (exists <datatype''>.
+          y : <datatype''> /\
+          <datatype''> <: <datatype> /\
+          <datatype''> <: <datatype> ~> f)
+       then ?(f y)
+       else null)
+```
+where
 ```
+v : <datatype>  iff  M(v : <datatype>) is defined
+```
+The last rule *optimistically* tries to decode an option value when the types are not statically known to match.
```

Comment:
There is no “when not statically known to match” due to “Simple Deserialisation: Deserialisation does not require (sub)type checks other than the type-directed traversal of the value blob”